subsys/bluetooth/services/fast_pair/fp_registration_data.c

```diff
@@ -103,15 +116,20 @@ static int fp_registration_data_validate(const struct device *unused)
 	ARG_UNUSED(unused);
 
 	int err;
+	uint8_t data[FP_DATA_SIZE];
 	const struct flash_area *fa = NULL;
 
 	err = flash_area_open(FP_PARTITION_ID, &fa);
 	if (!err) {
-		err = validate_fp_magic(fa, FP_MAGIC_HEADER_OFF);
+		err = flash_area_read(fa, FP_DATA_START, data, FP_DATA_SIZE);
+	}
+
+	if (!err) {
+		err = validate_fp_magic(&data[FP_OFFSET_TO_DATA_IDX(FP_MAGIC_OFF)]);
```

Comment:
This macro should be removed, because you already read from `FP_DATA_START` offset when you access FLASH